### PR TITLE
Custom.vs.sync.metadata

### PR DESCRIPTION
**Pull request by corinnekrych**
@wei-lee Fix the metaData issue introduced by: https://github.com/feedhenry/fh-ios-sdk/pull/42
FH-2262 related

### CODE REVIEW

File: fh-ios-sdk/Sync/FHSyncDataset.m [hunk 1]
```diff
@@ -383,7 +384,7 @@ - (void)startSyncTask:(NSDictionary *)info {
         syncLoopParams[@"fn"] = @"sync";
         syncLoopParams[@"dataset_id"] = self.datasetId;
         syncLoopParams[@"query_params"] = self.queryParams;
-        syncLoopParams[@"meta_data"] = self.metaData;
+        syncLoopParams[@"meta_data"] = self.syncMetaData;
```
Comment (wei-lee) [inline fh-ios-sdk/Sync/FHSyncDataset.m]: I believe this should be the customMetaData instead. This meta_data is sent to the cloud. The syncMetaData is only used locally and do not need to be sent to the cloud.

File: fh-ios-sdk/Sync/FHSyncDataset.m [hunk 2]
```diff
@@ -530,7 +531,7 @@ - (void)syncRecords {
     syncRecsParams[@"fn"] = @"syncRecords";
     syncRecsParams[@"dataset_id"] = self.datasetId;
     syncRecsParams[@"query_params"] = self.queryParams;
-    syncRecsParams[@"meta_data"] = self.metaData;
+    syncRecsParams[@"meta_data"] = self.syncMetaData;
```
Comment (wei-lee) [inline fh-ios-sdk/Sync/FHSyncDataset.m]: save as above

**Comment by wei-lee** [comment]
I think the JSONData method in FHSyncDataset should be updated to persist the syncMetaData to the files and read it back from files.

**Comment by wei-lee** [comment]
This is the method to [serialize the FHSyncData](https://github.com/feedhenry/fh-ios-sdk/blob/master/fh-ios-sdk/Sync/FHSyncDataset.m#L63), and this is the method to [deserialize](https://github.com/feedhenry/fh-ios-sdk/blob/master/fh-ios-sdk/Sync/FHSyncDataset.m#L123). We should make sure the syncMetaData is added to both.

**Comment by corinnekrych** [comment]
landed in 37fea14b901ff48777687cc06a728b3d48830701

**Pull request closed without merging**